### ISSUE TITLE
[ARPA] Add feature to export reporting data into a ZIP for gov partners auditing purposes

### ISSUE THREAD

**Issue by mindyatwork** [usdigitalresponse/usdr-gost#887]
Due to auditing requirements, gov partners need the ability to easily download all the Input templates that went into what they ultimately submit to Treasury.

### 💭 Proposal
- A button that downloads a Zip file of:
    - All the `xlsm` files that show up when you filter by `Only exported to treasury` for only the quarter the user is currently in.
    - An `uploaded_by.csv` that includes, per row:
        - filename
        - Email of the person who uploaded it
        - Agency
        - EC code
        - time stamp of upload
- Open design questions:
    - Where does this button sit? In the `Uploads` or `Dashboard` tab?
    - What is the text/color on the button?
    
 See full 1-pager here: https://www.notion.so/usdr/Export-reporting-data-into-a-CSV-828402124113453d8109e145e191eb85

**Comment by mhuang-usdr** [comment]
Andrew is working on this one

**Comment by agn-dsgn** [comment]
Hi! I'm the designer picking this up. Could the owner of the [one-pager](https://www.notion.so/usdr/Export-reporting-data-into-a-CSV-828402124113453d8109e145e191eb85) (linked above) please add me to it?

Not sure who to ask @mhuang-usdr - thanks!

**Comment by mhuang-usdr** [comment]
So odd, I thought I gave you access already! Sent it back over. Thanks for
asking :)
—
[image: USDR Logo]
Mindy Huang
Product Manager, Economic Stability
E ***@***.***
Discover new USDR volunteer opportunities on our Twitter
<https://twitter.com/USDResponse>, LinkedIn
<https://www.linkedin.com/company/usdr>, and Medium
<https://medium.com/u-s-digital-response>

U.S. Digital Response <https://www.usdigitalresponse.org/> is a nonprofit
that matches pro-bono technologists with government and organizations
responding to crisis, to quickly deliver critical services and
infrastructure that support the needs of the public. We’re non-partisan,
fast, and free.

**Comment by agn-dsgn** [comment]
Discussed design options with @caitlinwinner and landed on this one (shown below). Looking for feedback from @mhuang-usdr and the relevant engineer (who be you??)

**Overview:** The suggestion is to add a button within the table in `Uploads` that will allow users to download all the workbooks/uploads and an index .csv file 

**Button styling:**
- add blue button to the right of "reset filters"
- use same styling as "download file" within each table row (+ same hover state)
- button text: "`Download all as .zip"`

**Functionality:**
- on click, execute download of a .zip containing the following materials: a .xlsm of every filtered workbook and an index .csv file
- if there is no filter applied to the table, I suggest to still show the button to make for a more consistent UI and allow users to download all uploads (and all their versions) if desired
- index file includes this info for each row/workbook: filename, email of the person who uploaded it, agency, EC code, time stamp of upload

**Additional changes:**
- remove the question marks `?` after each of the existing filters


Questions:
- Any issues with allowing the button to persist with or without the filter? Could the larger data sets be too large / slow / problematic? Could this affect .zip file preparation time and necessitate a `downloading state`?
- What should we name the .index file? Ticket suggestion was `uploaded_by.csv,` and an alternate idea is `index_MM-DD-YYYY.csv`
- What should we name the .zip? Maybe something like `workbooks_MM-DD-YYYY`?

![ARPA-Reporter-887-01](https://user-images.githubusercontent.com/127222088/225106173-7c45bfeb-5d40-4de1-8c5e-03231b6b1eaf.png)

**Comment by agn-dsgn** [comment]
Great! The naming suggestions are perfect. Here are edited specs, with a couple final questions:

# Design Spec r2
**Overview:** We will add a button within the table in `Uploads` that will allow users to download all the workbooks/uploads and an index .csv file

**Button:**
- add button to the right of "reset filters"
- button text: "`Download all as .zip`"
- use same styling as "reset filters" button but with blue background, hex `#1565c0`
- hover state: same as for "Download file" button
- SEE BELOW FOR RELATED QUESTIONS

**Functionality:**
- on click, execute download of a .zip containing the following materials: a .xlsm of every filtered workbook and an index .csv file
- if no filter applied, include every workbook
- index file includes this info for each row/workbook: filename, email of the person who uploaded it, agency, EC code, time stamp of upload

**File naming:**
- .zip file: `workbooks_[reporting period name]_YYYY-MM-DD.zip`
- .csv index file: `index_[reporting period name]_YYYY-MM-DD.csv`

**Additional changes:**
- remove the question marks `?` after each of the existing filters

**Questions:**
- button progress indicator TBD: any existing pattern? Preference is to replace text with a spinner while .zip is preparing, maintaining full button width. Text returns when file has begun downloading
- button disabled state TBD: disable button if there are no uploads yet. Do we have a disabled state?
- error handling: Do we have a generic message or pattern if this action fails?

**Comment by TylerHendrickson** [comment]
@TylerHendrickson Note to self: This should probably be blocked by the S3 transition.

**Comment by ClaireValdivia** [comment]
@as1729 curious about where this fits in with the s3 work, and wondering if there needs to be two issues - one to export ALL historic files as a zip, and another to be able to download files from a single closed period.

**Comment by as1729** [comment]
As @TylerHendrickson has commented earlier here. We should probably follow this order:
- #2455 
- #2477 
- This issue

The 2477 and this issue can be worked in parallel once we have officially moved over to S3.

**Comment by ClaireValdivia** [comment]
2/7 convo - confirmed that we want two separate issues - one to export ALL historic files as a zip, and another to be able to download files from a single closed period.

**Comment by ClaireValdivia** [comment]
Closing as this is now a duplicate of #3803
